### PR TITLE
release-21.1: cli: Add cockroach dump support for --dump-mode=schema

### LINKED ISSUE #61756: sql/cli: make `cockroach dump --dump-mode=schema` delegate to SHOW CREATE ALL TABLES
We want to make sure the use case described in https://github.com/cockroachdb/cockroach/issues/53488 is not broken.

In 21.2, we should remove the `dump` CLI command entirely.

cc  @RichardJCai that means you don't need to add anything to `master` -- just the `release-21.1` branch.

### PR DESCRIPTION
**Pull request by RichardJCai**
Backport 2/2 commits from #61782.

/cc @cockroachdb/release

---

cli: Add cockroach dump support for --dump-mode=schema

Release note (cli change): Add back support for cockroach dump --dump-mode=schema.
This command calls SHOW CREATE ALL TABLES and returns the output.
This gives users a migration path to use in 21.1 while switching over
to SHOW CREATE ALL TABLES for 21.2.
When executing this, the user is warned that the command is being removed
in 21.2 and they should use SHOW CREATE ALL TABLES instead.

This is only intended to be backported to 21.1 and not added to master.

Release justification: Provide migration path for users who rely on cockroach dump --dump-mode=schema before killing off command entirely.

Resolves #61756

### CODE REVIEW

**Review by adityamaru — APPROVED** [review]

**Comment by adityamaru** [comment]
Oh actually, one nit before you merge.

File: pkg/sql/catalog/descriptor.go [hunk 1]
```diff
@@ -144,8 +144,8 @@ type TableDescriptor interface {
 	// AllIndexes returns a slice with all indexes, public and non-public,
 	// in the underlying proto, in their canonical order:
 	// - the primary index,
-	// - the public non-primary indexes in the Indexes array, in order,
-	// - the non-public indexes present in the Mutations array, in order.
+	//	// - the public non-primary indexes in the Indexes array, in order,
```
Comment (adityamaru) [inline pkg/sql/catalog/descriptor.go:147]: unintentional change?

**Review by rafiss — APPROVED** [review]
lgtm after fixing the formatting of that comment

File: pkg/sql/catalog/descriptor.go (hunk 1 above)
Comment (RichardJCai) [reply pkg/sql/catalog/descriptor.go:147]: Yep, good catch

**Pull request merged**